src/forms/take-ownership-alert.ts

```diff
@@ -1,81 +1,61 @@
-    const mattermostUrl: string = call.context.mattermost_site_url;
-    const botAccessToken: string = call.context.bot_access_token;
+    const actingUser: AppActingUser | undefined = call.context.acting_user;
     const username: string = call.context.acting_user.username;
-    const userId: string = call.context.acting_user.id;
     const values: AppCallValues | undefined = call.values;
     const i18nObj = configureI18n(call.context);
     const alertTinyId: string = typeof values?.[AckAlertForm.NOTE_TINY_ID] === 'undefined' ?
         call.state.alert.tinyId as string :
         values?.[AckAlertForm.NOTE_TINY_ID];
+    const apiKey = getOpsGenieAPIKey(call);
 
-    const mattermostOptions: MattermostOptions = {
-        mattermostUrl: <string>mattermostUrl,
-        accessToken: <string>botAccessToken,
-    };
-
-    const mattermostClient: MattermostClient = new MattermostClient(mattermostOptions);
-    const options: KVStoreOptions = {
-        mattermostUrl: <string>mattermostUrl,
-        accessToken: <string>botAccessToken,
-    };
-    const kvStoreClient = new KVStoreClient(options);
-
-    const kvConfig: ConfigStoreProps = await kvStoreClient.kvGet(StoreKeys.config);
     const opsGenieOpt: OpsGenieOptions = {
-        api_key: kvConfig.opsgenie_apikey,
+        api_key: apiKey,
     };
     const opsGenieClient = new OpsGenieClient(opsGenieOpt);
 
-    const mattermostUser: User = await mattermostClient.getUser(<string>userId);
-
     const identifierUser: Identifier = {
-        identifier: mattermostUser.email,
+        identifier: actingUser.email,
         identifierType: IdentifierType.USERNAME,
     };
 
-    await tryPromise(opsGenieClient.getUser(identifierUser), ExceptionType.MARKDOWN, i18nObj.__('forms.error'));
+    await tryPromise(opsGenieClient.getUser(identifierUser), ExceptionType.MARKDOWN, i18nObj.__('forms.error'), call.context.mattermost_site_url, call.context.app_path);
 
     const identifier: Identifier = {
         identifier: alertTinyId,
         identifierType: IdentifierType.TINY,
     };
-    const responseAlert: ResponseResultWithData<Alert> = await tryPromise(opsGenieClient.getAlert(identifier), ExceptionType.MARKDOWN, i18nObj.__('forms.error'));
-    const alert: Alert = responseAlert.data;
-    const alertURL: string = await getAlertLink(alertTinyId, alert.id, opsGenieClient);
+    const alert: Alert = await canUserInteractWithAlert(call, alertTinyId);
+    const alertURL: string = await getAlertLink(alertTinyId, alert.id, opsGenieClient, call.context.mattermost_site_url, call.context.app_path);
 
-    if (alert.owner === mattermostUser.email) {
-        throw new Exception(ExceptionType.MARKDOWN, i18nObj.__('forms.actions.exception-owner', { alert: alertURL }));
+    if (alert.owner === actingUser.email) {
+        throw new Exception(ExceptionType.MARKDOWN, i18nObj.__('forms.actions.exception-owner', { alert: alertURL }), call.context.mattermost_site_url, call.context.app_path);
     }
 
```

Comment:
Missing check `canUserInteractWithAlert`

Reply:
The check is in line `44`, before performing any action to an alert, we validate with:
`const alert: Alert = await canUserInteractWithAlert(call, alertTinyId);`